### PR TITLE
GitHub hook fixes

### LINKED ISSUE #4180: Use hmac.compare_digest in the GitHub webhook if possible
Currently buildbot uses naive `!=` to compare digests. 

https://github.com/buildbot/buildbot/blob/f522a22a88d6f2d39e8ff1afe639dc7ff6ef4a39/master/buildbot/www/hooks/github.py#L117-L120

I propose to use hmac.compare_digest() if possible. For example: (not tested yet)

```Python
def _cmp(a, b):
    try:
        # try the more secure compare_digest() first
        from hmac import compare_digest
        return compare_digest(a, b)
    except ImportError:
        # and fallback to the insecure simple comparison otherwise
        return a == b

if not _cmp(mac.hexdigest(), hexdigest):
    raise ValueError('Hash mismatch') 
```

If this sounds a good idea, I'll try to open a pull request.

BTW, HMAC verification has been implemented long time ago (https://github.com/buildbot/buildbot/commit/baf1f47a5521dae8752cc4a236cbf35ff4db0e18), but the doc implies it's missing (https://github.com/buildbot/buildbot/blob/master/master/docs/manual/cfg-wwwhooks.rst)

> Patches are welcome to implement: https://developer.github.com/webhooks/securing/

EDIT: a grammar fix

### PR DESCRIPTION
**Pull request by yan12125**
## Contributor Checklist:

* [ ] I have updated the unit tests
* [ ] I have created a file in the `master/buildbot/newsfragment` directory (and read the `README.txt` in that directory)
     - Not sure if such a change is worth a news item or not
* [x] I have updated the appropriate documentation

Closes #4180

### CODE REVIEW

**Comment by yan12125** [comment]
Buildbot failures are due to network issues:
```
Cloning into '.'...
error: RPC failed; curl 56 GnuTLS recv error (-54): Error in the pull function.
fatal: The remote end hung up unexpectedly
fatal: early EOF
fatal: index-pack failed
```
(https://nine.buildbot.net/#/builders/6/builds/2573)
```
error: RPC failed; curl 56 GnuTLS recv error (-54): Error in the pull function.
fatal: The remote end hung up unexpectedly
fatal: early EOF
fatal: index-pack failed
```
https://nine.buildbot.net/#/builders/5/builds/2625

And the Appveyor job is pending in a queue with more than 100 jobs.

BTW, should I squash the follow-up fixes or keep them as-is for now?

File: master/buildbot/www/hooks/github.py [hunk 1]
```diff
@@ -114,9 +114,17 @@ def _get_payload(self, request):
...
                            msg=unicode2bytes(content),
                            digestmod=sha1)
-            # NOTE: hmac.compare_digest should be used, but it's only available
-            # starting Python 2.7.7
-            if mac.hexdigest() != hexdigest:
+
+            def _cmp(a, b):
+                try:
+                    # try the more secure compare_digest() first
+                    from hmac import compare_digest
```
Comment (tardyp) [inline master/buildbot/www/hooks/github.py]: we don't support python 2.6, so that's fine to always use  compare_digest

Comment (yan12125) [reply master/buildbot/www/hooks/github.py]: Hmm. Are Python 2.7.0~2.7.6 unsupported, either?

Comment (tardyp) [reply master/buildbot/www/hooks/github.py]: well, we don't go to that precision, indeed. But we don't test python minor versions, so I don't . I guess lets keep it that way, as it is not much of a burden. Stats show that there are around 111 unique installs with python 2.7.5..

**Pull request merged**